### PR TITLE
Stop the watchdog timer in error case of sem_wait failure

### PR DESCRIPTION
**Pull request by pradeepns85**
Since wdog_timer has been started before waiting for semaphore and
if Sem_wait fails, wdog timer should be canceled before attempting
for delete operation. Though wdog_delete API takes care of canceling
an active timer, One must consider canceling the timer as soon as
possible to avoid calling wdog timeout handler and overwrite the
sem_wait error value as ETIMEOUT .

Signed-off-by: pradeep.ns <pradeep.ns@samsung.com>

### CODE REVIEW

File: os/kernel/semaphore/sem_tickwait.c [hunk 1]
```diff
@@ -183,6 +183,11 @@ int sem_tickwait(FAR sem_t *sem, systime_t start, uint32_t delay)
 		/* Return the errno from sem_wait() */
 
 		ret = -get_errno();
+
+		/* Stop the watchdog timer as it's started before sem_wait */
+
+		wd_cancel(rtcb->waitdog);
```
Comment (sunghan-chang) [inline os/kernel/semaphore/sem_tickwait.c]: how about remove this and below line (goto errout_with_irqdisabled) to reduce code size? Is it not good for readability and posibility of adding new code?

Comment (pradeepns85) [inline os/kernel/semaphore/sem_tickwait.c]: Ok. Done. Thanks

**Comment by JunhwanPark** [comment]
@pradeepns85 
Hi,  I have a question.
Do you know why many patches have recently been removed from the master branch and are included in your pull request?

**Comment by heesub** [comment]
@pradeepns85 You need to rebase your topic-branch on top of latest master and then send a new pull request.

**Comment by sunghan-chang** [comment]
Removed commits are recovered. Please rebase yours.

**Pull request closed without merging**